cmake/Platform/Common/Install_common.cmake

```diff
@@ -349,6 +349,133 @@ function(ly_setup_subdirectories)
+        endif()
+
+        list(APPEND copy_dependencies_target_rel_path "${destination_relpath}")
+    endforeach()
+
+    # sanity check
+    if (copy_dependencies_source_file_path OR copy_dependencies_target_rel_path)
+        list(LENGTH copy_dependencies_source_file_path source_list_length)
+        list(LENGTH copy_dependencies_target_rel_path  relpath_list_lenth)
+        if (NOT source_list_length EQUAL relpath_list_lenth)
+            message(FATAL_ERROR "Programmer error - for target ${ALIAS_TARGET_NAME}, lists are supposed to be same length, but\n\
+                                ${source_list_length} source and ${relpath_list_lenth} relpath\n\
+                                source:'${copy_dependencies_source_file_path}'\n\
+                                relpath: '${copy_dependencies_target_rel_path}'")
+        endif()
+    endif()
+
+    unset(${OUTVAR_GENERATED_DATA_FILENAME} PARENT_SCOPE)
+
+    # the absolute source paths of files to copy can have generator expressions that are per-config.
+    # use file(GENERATE ...) to evaluate the genexes, by writing it as a build file
+    # since GENERATE only happens during generate step, this is as far as we can go during
+    # configure.  It will be up to a build step to read these generated files back and
+    # do a final transform that removes things like absolute paths.
+    # note that we generate a file for every 3p target, as they must exist.
+    set(generate_content "set(target_name ${ALIAS_TARGET_NAME})\n")
+    if (copy_dependencies_source_file_path)
+        string(APPEND generate_content "set(copy_dependencies_source_file_path \"${copy_dependencies_source_file_path}\")\n")
+        string(APPEND generate_content "set(copy_dependencies_target_rel_path \"${copy_dependencies_target_rel_path}\")\n")
+    endif()
+
+    file(GENERATE OUTPUT  "${CMAKE_BINARY_DIR}/install/$<CONFIG>/${PAL_PLATFORM_NAME}/${CLEAN_TARGET_NAME}.cmake" 
+                  CONTENT "${generate_content}")
+    set(${OUTVAR_GENERATED_DATA_FILENAME} "${CMAKE_BINARY_DIR}/install/$<CONFIG>/${PAL_PLATFORM_NAME}/${CLEAN_TARGET_NAME}.cmake" PARENT_SCOPE)
+    set(${OUTVAR_FILE_NAME_TO_GENERATE} ${CLEAN_TARGET_NAME} PARENT_SCOPE)
+
+endfunction()
+
+# In "Script only mode", the 3p libraries are not downloaded at all.
+# But some of them (not all) have extraneous .so or .dll files that need to be deployed
+# into the final runtime package if you intend to ship a script-only runtime.
+# This function exports a file that is used only in script-only mode to supply that list
+# of extra DLLs to copy along with the game.
+function(ly_setup_3p_dependencies)
+    unset(list_of_files_to_process)
+    unset(expected_output_files)
+    set(final_3p_output_dir "${CMAKE_BINARY_DIR}/cmake/3rdParty/Platform/${PAL_PLATFORM_NAME}/${LY_BUILD_PERMUTATION}")
+    get_property(o3de_all_3rdparty_targets GLOBAL PROPERTY O3DE_ALL_3RDPARTY_TARGETS)
+    list(REMOVE_DUPLICATES o3de_all_3rdparty_targets)
```

Comment:
There shouldn't be any duplicates, since `LyWrappers.cmake` checks that only a 3rdParty target that doesn't exist is added to the global property